### ISSUE TITLE
[Research] Add CNN-KAN-F2CA Sparse-Channel Emotion Model (89% DEAP 4-class, 4 channels)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#65]
## What
Train a CNN-KAN (Kolmogorov-Arnold Network) emotion classifier that encodes DE + DASM/RASM features from 4 channels into pseudo-RGB images for spatial-temporal CNN processing. Replaces traditional dense layers with KAN for better generalization on sparse channels.

## Why
- PLOS ONE (2025): CNN-KAN-F2CA achieves **89.16% on DEAP 4-class** with only 4 channels (subject-dependent) and **97.45% on SEED 3-class**
- Directly applicable — Muse 2 has exactly 4 channels; DE + DASM/RASM features already computed in `eeg_processor.py`
- KAN layers learn activation functions per-edge rather than fixed ReLU — better at capturing nonlinear EEG patterns
- Cross-subject 4-channel: expected 70-80% — potentially above current 74.21% LightGBM

### Step 1: Create pseudo-image encoder
```python
# ml/models/cnn_kan_emotion.py
import torch
import torch.nn as nn

class PseudoImageEncoder:
    def encode(self, features_multichannel):
        # Reshape (4 channels × 5 bands × 3 feature types) → (3, 4, 5) pseudo-RGB
        # Channel 0 (R): DE features per channel per band
        # Channel 1 (G): PSD features per channel per band
        # Channel 2 (B): Asymmetry features (DASM/RASM)

class CNNKAN(nn.Module):
    def __init__(self, n_classes=6):
        # Conv2D layers for spatial-spectral patterns
        # KAN layers replace final dense layers
        # Output: 6-class emotion probabilities
```

### Step 2: Train on existing datasets
- Use DEAP + DREAMER + EmoKey features already extracted
- Pseudo-image encoding of existing 85-dim features
- Export to ONNX for inference

### Step 3: Add as prediction path
- Insert in emotion_classifier.py fallback chain above LightGBM
- Activate when ONNX model file exists

### Files to Modify
- `ml/models/cnn_kan_emotion.py` (NEW) — CNN-KAN architecture + pseudo-image encoding
- `ml/training/train_cnn_kan.py` (NEW) — training script
- `ml/models/emotion_classifier.py` — add CNN-KAN prediction path

### Expected Impact
- Cross-subject 4-channel: 70-80% binary valence/arousal
- Better than LightGBM for capturing spatial-spectral interactions
- ONNX export enables mobile inference

## Complexity
MEDIUM — standard CNN + KAN layers in PyTorch, feature pipeline already exists

## References
- [CNN-KAN-F2CA sparse channel emotion recognition](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0322583) — PLOS ONE, 2025
